Lib/test/test_dataclasses.py

```diff
@@ -192,6 +192,20 @@ class C:
         first = next(iter(sig.parameters))
         self.assertEqual('self', first)
 
+    def test_field_named_object(self):
+        @dataclass
+        class C:
+            object: str
+        c=C('foo')
+        self.assertEqual(c.object, 'foo')
+
+    def test_field_named_object_frozen(self):
+        @dataclass(frozen=True)
+        class C:
+            object: str
+        c=C('foo')
+        self.assertEqual(c.object, 'foo')
+
```

Comment:
In addition to these tests, please add a test for field names that are all identifiers in `__builtins__` which don't start with double underscores, except for `None`, `True`, and `False`. You can use `make_dataclass` to dynamically generate a class with those members. I think this will work to generate the field names: `[n for n in __builtins__.__dict__.keys() if not n.startswith('__') and not n in ('None', 'False', 'True')]`. You'll want to do this for both a normal dataclass and a frozen one. Also, you'll want to test `order=True`.

Reply:
Something like:
```
names = [n for n in __builtins__.__dict__.keys() if not n.startswith('__') and not n in ('None', 'False', 'True')]
C = make_dataclass('C', names)
C(*([0] * len(names)))
```
Which works in 3.7.0, and
```
names = [n for n in __builtins__.__dict__.keys() if not n.startswith('__') and not n in ('None', 'False', 'True')]
C = make_dataclass('C', names)
C(*([0] * len(names)))
```
Which fails in 3.7.0.